src/modules/utils.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+/**
+ * A function to generate a pause in promise chaining
+ * @param  {number} timer How much ws to wait
+ * @return {Promise}      A Promise that will resolve when timer is expired
+ */
+module.exports.timeoutPromise = (timer) => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, timer);
```

Comment:
Can you provide a sample of your implementation of this delay promise with nextTick?